### PR TITLE
Adding postcode lookup pattern

### PR DESCRIPTION
**Pull request by chris-gds**
Trello: https://trello.com/c/6mYZd3HQ/830-create-postcode-lookup-component

Adding local postcode lookup pattern with additional styling, stripping tracking markup

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/finder-frontend), after merging.

### CODE REVIEW

File: app/views/components/_postcode-lookup.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<div class="postcode-lookup" data-module="track-submit"
+  <% if @location_error %>
+    <%
+      if @location_error.sub_message.present?
+        description = t(@location_error.sub_message)
+      end
+    %>
+    <%= render "govuk_publishing_components/components/error_alert", {
+      message: t(@location_error.message, **@location_error.message_args),
```
Comment (chris-gds) [inline app/views/components/_postcode-lookup.html.erb:9]: Aware these are not defined yet - can strip if more helpful?

File: app/assets/stylesheets/components/_postcode-lookup.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.postcode-lookup {
+  padding: govuk-spacing(3);
+  margin: govuk-spacing(6) 0;
+  background: govuk-colour("light-grey", $legacy: "grey-4");
```
Comment (DilwoarH) [inline app/assets/stylesheets/components/_postcode-lookup.scss:4]: legacy support not needed, should be removed.

File: app/views/components/_postcode-lookup.html.erb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<div class="postcode-lookup" data-module="track-submit"
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:1]: this div does not end, also is the track submit module working?

File: app/views/components/_postcode-lookup.html.erb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  <% if @location_error %>
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:2]: Where is this variable defined?

File: app/views/components/_postcode-lookup.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      if @location_error.sub_message.present?
+        description = t(@location_error.sub_message)
+      end
+    %>
+    <%= render "govuk_publishing_components/components/error_alert", {
+      message: t(@location_error.message, **@location_error.message_args),
+      description: description,
+    } %>
+  <% end %>
+  <form method="post" id="local-locator-form">
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:13]: the id does not seem to be needed for anything?

File: app/views/components/_postcode-lookup.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    <fieldset>
+      <legend class="visuallyhidden">Postcode lookup</legend>
+      <%= render "govuk_publishing_components/components/input", {
+          label: {
+            text: "Enter a postcode"
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:18]: This should be passed into the component

File: app/views/components/_postcode-lookup.html.erb [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<div class="postcode-lookup" data-module="track-submit"
+  <% if @location_error %>
+    <%
+      if @location_error.sub_message.present?
+        description = t(@location_error.sub_message)
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:5]: variable needs to be defined

File: app/views/components/_postcode-lookup.html.erb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    } %>
+  <% end %>
+  <form method="post" id="local-locator-form">
+    <fieldset>
+      <legend class="visuallyhidden">Postcode lookup</legend>
+      <%= render "govuk_publishing_components/components/input", {
+          label: {
+            text: "Enter a postcode"
+          },
+          value: postcode,
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:20]: same as above

File: app/views/components/_postcode-lookup.html.erb [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+          name: "postcode",
+          id: "postcode",
+          hint: "For example SW1A 2AA",
+          invalid: @location_error ? 'true' : 'false',
+          autocomplete: "postal-code",
+        } %>
+      <%= render "govuk_publishing_components/components/button", text: "Find", margin_bottom: true %>
```
Comment (DilwoarH) [inline app/views/components/_postcode-lookup.html.erb:27]: text should be passed in

**Review by DilwoarH — CHANGES_REQUESTED** [review]

**Pull request closed without merging**